test/functional/test_short_deck.py

```diff
@@ -221,6 +231,148 @@ def test_call_action_sequence(n_players):
+                action_sequence = betting_stage_dict[betting_stage]["action_sequence"]
+                possible_sequences = action_sequences[n_players_started_round]
+
+                assert action_sequence in possible_sequences
+
+
+def test_skips(n_players: int = 3):
+    """
+    Check each round to make sure that skips are mod number of players and appended on
+    the skipped player's turn
+    """
+    # Seed the random number generation so things are procedural/reproducable.
+    seed(42)
+    for _ in range(500):
+        state, _ = _new_game(n_players=n_players, small_blind=50, big_blind=100)
+
+        betting_stage_dict = {
+            "pre_flop": None,
+            "flop": None,
+            "turn": None,
+            "river": None,
+        }
+
+        first_flop_action = True
+        first_turn_action = True
+        first_river_action = True
+
+        betting_stage = "pre_flop"
+        previous_betting_stage = None
+        while True:
+            if betting_stage != state.betting_stage:
+                previous_betting_stage = betting_stage
+                betting_stage = state.betting_stage
+
+            uniform_probability: float = 1 / len(state.legal_actions)
+            probabilities = np.full(len(state.legal_actions), uniform_probability)
+            random_action: str = np.random.choice(state.legal_actions, p=probabilities)
+
+            state = state.apply_action(random_action)
+
+            if state.betting_stage == "flop" and first_flop_action:
+                betting_stage_dict["pre_flop"] = state._history
+                first_flop_action = False
+                flop_start = len(betting_stage_dict["pre_flop"])
+            if state.betting_stage == "turn" and first_turn_action:
+                betting_stage_dict["flop"] = state._history[flop_start:]
+                first_turn_action = False
+                turn_start = len(betting_stage_dict["flop"]) + flop_start
+            if state.betting_stage == "river" and first_river_action:
+                betting_stage_dict["turn"] = state._history[turn_start:]
+                first_river_action = False
+                river_start = len(betting_stage_dict["turn"]) + turn_start
+            if state.betting_stage in {"show_down", "terminal"}:
+                if previous_betting_stage == None:
+                    betting_stage_dict["pre_flop"] = state._history
+                elif previous_betting_stage == "pre_flop":
+                    betting_stage_dict["flop"] = state._history[flop_start:]
+                elif previous_betting_stage == "flop":
+                    betting_stage_dict["turn"] = state._history[turn_start:]
+                elif previous_betting_stage == "turn":
+                    betting_stage_dict["river"] = state._history[river_start:]
+                break
+
+        # TODO: wrap this in a for loop, add history as property of poker state class
```

Comment:
I was also thinking that it might be nice to break up the history list, into a dict of lists (one for each betting stage). Would certainly help readability

Reply:
Yes, I am not happy with the succinctity of this skip test I wrote TBH. As a short term solution I would be more comfortable still using state._history for info sets, but having a method that would parse out the history per round as I have done, although it would add some inefficiencies CFR, so I wouldn't prioritize that.